charts/pelorus/templates/prometheus-rules.yaml

```diff
@@ -20,7 +20,7 @@ spec:
         avg(sdp:lead_time:by_app)
     - record: sdp:time_to_restore:by_issue
       expr: >
-        failure_resolution_timestamp{app!="unknown"} - failure_creation_timestamp{app!="unknown"}
+        failure_resolution_timestamp{app!="unknown"} - max_over_time(failure_creation_timestamp{app!="unknown"}[5y])
```

Comment:
Instead of 5 years I would like to use a dynamic value. We could use the value user sent from `prometheus_retention`, or even `$__range` from Grafana (but then we would not be able to store the rule here).

 But to be honest, I think I do not understand this metric. What should appear in selected ranges of Grafana: average of time to close issues that were created and resolved in that time range or it just need to be resolved inside that time range to appear?
 
I would like to have new tests in `_test/prometheus/test.yaml` to validate the changes made

Reply:
This works slightly differently.

The time range from Grafana is applied for the entire `sdp:time_to_restore:by_issue` expression, without even specifying `$__range`. The `A-B` is for the time series within the `$__range`, so both **A** and **B** needs to be there. Now think of a situation that **B** was created before the `$__range` in this situation the `A-B` will not work even if the range is within **A** timestamp. To make sure we can properly calculate `A-B` where **A** is **failure_resolution_timestamp** and **B** is **failure_creation_timestamp**, we need to ensure the **B** is available at the time when **A** have happened. For this we use **max_over_time** which will make **B** available from the time it was created until **now** but not further then 5y ago. We do not have knowledge when the **B** occurred, so we can't have dynamic range here, but it's safe to say not older then 5y timeframe.